packages/expo-modules-test-core/src/mockgen.ts

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env node
+'use strict';
+
+import fs from 'fs';
+import path from 'path';
+import ts from 'typescript';
+
+import { Closure, ClosureTypes, OutputModuleDefinition } from './types';
+
+const directoryPath = process.cwd();
+
+function maybeUnwrapSwiftArray(type: string) {
+  const isArray = type.startsWith('[') && type.endsWith(']');
+  if (!isArray) {
+    return type;
+  }
+  const innerType = type.substring(1, type.length - 1);
+  return innerType;
+}
+
+function isSwiftArray(type: string) {
+  return type.startsWith('[') && type.endsWith(']');
+}
+
+function mapSwiftTypeToTsType(
+  type: string
+): ts.KeywordTypeNode | ts.TypeReferenceNode | ts.ArrayTypeNode {
+  if (!type) {
+    return ts.factory.createKeywordTypeNode(ts.SyntaxKind.VoidKeyword);
+  }
+  if (isSwiftArray(type)) {
+    return ts.factory.createArrayTypeNode(mapSwiftTypeToTsType(maybeUnwrapSwiftArray(type)));
+  }
+  switch (type) {
+    case 'unknown':
+      return ts.factory.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword);
+    case 'String':
+      return ts.factory.createKeywordTypeNode(ts.SyntaxKind.StringKeyword);
+    case 'Bool':
+      return ts.factory.createKeywordTypeNode(ts.SyntaxKind.BooleanKeyword);
+    case 'Int':
+    case 'Float':
+    case 'Double':
+      return ts.factory.createKeywordTypeNode(ts.SyntaxKind.NumberKeyword);
+    default:
+      return ts.factory.createTypeReferenceNode(type);
```

Comment:
Is it like `any`? We support much more types on the Swift side, so we should extend the list of types here and have a fallback to `any` in other cases.